### PR TITLE
Fixed eeprom drivers

### PR DESCRIPTION
**Pull request by Sabramz**
EEPROM drivers now work with addresses instead of pages.

### CODE REVIEW

File: general/src/m24c32.c [hunk 1]
```diff
@@ -1,102 +1,62 @@
 #include "m24c32.h"
 
-uint16_t bytes_to_write(uint16_t size, uint16_t offset)
+uint16_t check_size(uint16_t size)
 {
-    if((size + offset) < M24C32_PAGE_SIZE) return size;
-    else return M24C32_PAGE_SIZE - offset;
```
Comment (dyldonahue) [inline general/src/m24c32.c:6]: i dont think we need a max page size anymore. Technically we need to not overflow eeprom, but its so big that i think we can ignore it for now

File: general/src/m24c32.c [hunk 2]
```diff
@@ -1,102 +1,62 @@
...
     HAL_StatusTypeDef status;
 
-    uint8_t start_page = page;
-    uint8_t end_page = page + ((size + offset) / M24C32_PAGE_SIZE);
-    uint8_t num_pages = end_page - start_page + 1;
-
-    uint8_t data_pos = 0;
-
-    for(int i = 0; i < num_pages; i++)
+    for(int bytes_written = 0; bytes_written < size; bytes_written += 32)
```
Comment (dyldonahue) [inline general/src/m24c32.c:13]: why are we jumping 32 bytes? should be writing byte by byte

File: general/src/m24c32.c [hunk 3]
```diff
@@ -1,102 +1,62 @@
...
-        offset = 0;
-        size = size - bytes_remaining;
     }
 
     return HAL_OK;
-}
-
-
-// TODO: Test this function, not sure if it will work
-HAL_StatusTypeDef eeprom_page_erase(uint8_t page)
```
Comment (Sabramz) [inline general/src/m24c32.c:96]: We don't have pages anymore, so I got rid of page_erase.

File: general/src/m24c32.c [hunk 4]
```diff
@@ -1,102 +1,62 @@
...
-
-    for(int i = 0; i < num_pages; i++)
+    for(int bytes_written = 0; bytes_written < size; bytes_written += 32)
     {
-        uint16_t mem_addr = start_page << 4 | offset;
-        uint16_t bytes_remaining = bytes_to_write(size, offset);
-
-        status = HAL_I2C_Mem_Write(i2c_handle, M24C32_I2C_ADDR, mem_addr, 2, &data[data_pos], 
-                                   bytes_remaining, 1000);
+        uint16_t write_size = check_size(size - bytes_written);
```
Comment (dyldonahue) [inline general/src/m24c32.c:15]: not sure i understand what we need check size for

File: general/src/m24c32.c [hunk 5]
```diff
@@ -1,102 +1,62 @@
...
     HAL_StatusTypeDef status;
 
-    uint8_t start_page = page;
-    uint8_t end_page = page + ((size + offset) / M24C32_PAGE_SIZE);
-    uint8_t num_pages = end_page - start_page + 1;
-
-    uint8_t data_pos = 0;
-
-    for(int i = 0; i < num_pages; i++)
+    for(int bytes_read = 0; bytes_read < size; bytes_read += 32)
```
Comment (dyldonahue) [inline general/src/m24c32.c:30]: same here as above, why +=32

File: general/src/m24c32.c [hunk 6]
```diff
@@ -1,102 +1,62 @@
 #include "m24c32.h"
 
-uint16_t bytes_to_write(uint16_t size, uint16_t offset)
+uint16_t check_size(uint16_t size)
```
Comment (Sabramz) [inline general/src/m24c32.c:3]: Max read and write size is 32 bytes in one call, so if size > 32, return 32.

File: general/src/m24c32.c (hunk 5 above)
Comment (dyldonahue) [reply general/src/m24c32.c:30]: ok i get it now, but think it unnecesarry. I think we should just read the number of bytes we want to, no need to account or 32 byte pages and read seperately

File: general/src/m24c32.c (hunk 2 above)
Comment (Sabramz) [inline general/src/m24c32.c:13]: Write in 32 byte batches. This is the same for the other functions.

File: general/src/m24c32.c (hunk 4 above)
Comment (Sabramz) [inline general/src/m24c32.c:15]: This is the size of the current write. Same for all functions.

**Review by dyldonahue — CHANGES_REQUESTED** [review]

File: middleware/src/eepromdirectory.c [hunk 7]
```diff
@@ -0,0 +1,90 @@
+#include "eepromdirectory.h"
+#include "general/include/m24c32.h"
```
Comment (Sabramz) [inline middleware/src/eepromdirectory.c:2]: Does this file have access to m24c32.h?

Comment (dyldonahue) [reply middleware/src/eepromdirectory.c:2]: now it does

**Review by dyldonahue — APPROVED** [review]

**Pull request merged**